### PR TITLE
Second challenge

### PR DESCRIPTION
**Pull request by OfekPoraz**

### CODE REVIEW

File: src/main/java/EncryptionAlgorithms/DoubleEncryption.java [hunk 1]
```diff
@@ -14,17 +15,17 @@ public DoubleEncryption(EncryptionAlgorithm firstAlgorithm, EncryptionAlgorithm
         super("DoubleEncryption");
         this.algorithm1 = firstAlgorithm;
         this.algorithm2 = secondAlgorithm;
-        this.secondKey = (new Key("DoubleEncryption")).getKey();
+        this.secondKey = (new Key()).getKey();
     }
 
     @Override
-    public String encryptFile(String stringToEncrypt, int encryptionKey) throws IOException {
+    public String encryptFile(String stringToEncrypt, int encryptionKey) throws IOException, NoSuchElementException, ClassCastException {
```
Comment (ido9000) [inline src/main/java/EncryptionAlgorithms/DoubleEncryption.java:22]: you not encrypt file here - change function name to just encrypt or encryptString etc.
*** in All Algorithems ***

File: src/main/java/EncryptionAlgorithms/DoubleEncryption.java [hunk 2]
```diff
@@ -14,17 +15,17 @@ public DoubleEncryption(EncryptionAlgorithm firstAlgorithm, EncryptionAlgorithm
...
         String middleEncryption = algorithm1.encryptFile(stringToEncrypt, encryptionKey);
         return algorithm2.encryptFile(middleEncryption, secondKey);
     }
 
     @Override
-    public String decryptFile(String stringToDecrypt, int decryptionKey) throws IOException {
+    public String decryptFile(String stringToDecrypt, int decryptionKey) throws IOException, NoSuchElementException, ClassCastException {
```
Comment (ido9000) [inline src/main/java/EncryptionAlgorithms/DoubleEncryption.java:28]: you not decrypt file here - change function name to just decrypt or decryptString etc.
*** in All Algorithems ***

File: src/main/java/EncryptionAlgorithms/ShiftMultiplyEncryption.java [hunk 3]
```diff
@@ -9,7 +10,7 @@ public ShiftMultiplyEncryption() {
     }
 
     @Override
-    public String encryptFile(String stringToEncrypt, int encryptionKey) throws IOException {
+    public String encryptFile(String stringToEncrypt, int encryptionKey) throws IOException, NoSuchElementException, ClassCastException{
         StringBuffer encryptedData = new StringBuffer();
         for (int i = 0 ; i < stringToEncrypt.length() ; i ++){
             if (checkIfAlphabet(stringToEncrypt.charAt(i), false)){
```
Comment (ido9000) [inline src/main/java/EncryptionAlgorithms/ShiftMultiplyEncryption.java:16]: you can use Character.isUpperCase(c) instead

** change in all algorithems**

File: src/main/java/FileEncryptor/FileEncryptor.java [hunk 4]
```diff
@@ -29,20 +29,35 @@ public FileEncryptor(EncryptionAlgorithm algorithm, String pathToFile) throws IO
...
         int key = Integer.parseInt(keyFile.readFromFile().strip());
         String dataToDecrypt = encryptedFile.readFromFile();
         String decryptedData = algorithm.decryptFile(dataToDecrypt, key);
         decryptedFile.writeToFile(decryptedData);
     }
+
+    public String getPathToDecryptedFile(){
+        return this.decryptedFile.getPathToFile();
+    }
+
```
Comment (ido9000) [inline src/main/java/FileEncryptor/FileEncryptor.java:50]: why not to use comparable interface and compareTo function in FileEncryptor class ( not inner class)?

File: src/main/java/Utils/FileOperations.java [hunk 5]
```diff
@@ -59,5 +62,14 @@ public String readFromFile() throws FileNotFoundException {
         return stringBuffer.toString();
     }
 
+    public boolean compareFiles(String pathToFileOne, String pathToFileTwo) throws IOException, NullPointerException {
```
Comment (ido9000) [inline src/main/java/Utils/FileOperations.java:65]: compare files by what?

File: src/test/java/DoubleEncryptionTest.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class DoubleEncryptionTest {
+
+    static final private String originalPath = "C:\\Users\\ofeko\\IdeaProjects\\EncryptionProject\\src\\main\\resources\\Hi.txt";
+    private final FileOperations fileOperations = new FileOperations(originalPath);
+    private FileEncryptor fileEncryptor;
+
+    public DoubleEncryptionTest() throws IOException {
```
Comment (ido9000) [inline src/test/java/DoubleEncryptionTest.java:18]: before annotation?

File: src/test/java/DoubleEncryptionTest.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import EncryptionAlgorithms.ShiftMultiplyEncryption;
+import EncryptionAlgorithms.ShiftUpEncryption;
+import FileEncryptor.FileEncryptor;
+import Utils.FileOperations;
+
+import java.io.IOException;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
```
Comment (ido9000) [inline src/test/java/DoubleEncryptionTest.java:11]: check also edge cases

File: src/test/java/ShiftUpTest.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class ShiftUpTest {
+
+    static final private String originalPath = "C:\\Users\\ofeko\\IdeaProjects\\EncryptionProject\\src\\main\\resources\\Hi.txt";
+    private final FileOperations fileOperations = new FileOperations(originalPath);
+    private FileEncryptor fileEncryptor;
+
+    public ShiftUpTest() throws IOException {
```
Comment (ido9000) [inline src/test/java/ShiftUpTest.java:17]: use try catch block ( if you use throws you send the exception to where?)

File: src/test/java/TestAll.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+
+
```
Comment (ido9000) [inline src/test/java/TestAll.java:4]: Very nice! excellent!:)

**Review by ido9000 — CHANGES_REQUESTED** [review]
